From 13b297256867d19bcf2c8c1341acbb2009ff3a39 Mon Sep 17 00:00:00 2001
From: LordPsyan <uppp@juno.com>
Date: Sun, 27 May 2012 06:46:41 -0500
Subject: [PATCH] 2012_05_27-Enable-Emerald-Forest-core

---
 src/game/Player.cpp |   21 ++++++++++++---------
 1 files changed, 12 insertions(+), 9 deletions(-)

diff --git a/src/game/Player.cpp b/src/game/Player.cpp
index f6e019b..eea660d 100644
--- a/src/game/Player.cpp
+++ b/src/game/Player.cpp
@@ -1733,21 +1733,24 @@ bool Player::TeleportTo(uint32 mapid, float x, float y, float z, float orientati
 
     if (GetMapId() != mapid)
     {
-        if (!(options & TELE_TO_CHECKED))
+        if (mapid != 169)
         {
-            if (!CheckTransferPossibility(mapid))
+            if (!(options & TELE_TO_CHECKED))
             {
-                if (GetTransport())
-                    TeleportToHomebind();
+                if (!CheckTransferPossibility(mapid))
+                {
+                    if (GetTransport())
+                        TeleportToHomebind();
 
-                DEBUG_LOG("Player::TeleportTo %s is NOT teleported to map %u (requirements check failed)", GetName(), mapid);
+                    DEBUG_LOG("Player::TeleportTo %s is NOT teleported to map %u (requirements check failed)", GetName(), mapid);
 
-                return false;                                       // normal client can't teleport to this map...
+                    return false;                                       // normal client can't teleport to this map...
+                }
+                else
+                    options |= TELE_TO_CHECKED;
             }
-            else
-                options |= TELE_TO_CHECKED;
+            DEBUG_LOG("Player::TeleportTo %s is being far teleported to map %u", GetName(), mapid);
         }
-        DEBUG_LOG("Player::TeleportTo %s is being far teleported to map %u", GetName(), mapid);
     }
     else
     {
-- 
1.7.2.5

